analyzer/analyzer.go

```diff
@@ -158,8 +160,11 @@ func (p *analyzer) Close() (err error) {
 	if p.tailer != nil {
 		err = p.tailer.Close()
 	}
-	if err2 := os.RemoveAll(p.logdir); err == nil {
-		err = err2
+
+	if p.config.WorkDir == "" {
+		if err2 := os.RemoveAll(p.wd); err == nil {
```

Comment:
Wrap the other `os.RemoveAll` calls like this.

Reply:
All the other ones are called when an error is returned and so its part of an abort. Not interested in the error it would return.

Reply:
Oh nvm, you are right!